nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java

```diff
@@ -62,27 +65,45 @@ private NullabilityUtil() { }
      */
     public static Symbol.MethodSymbol getFunctionalInterfaceMethod(LambdaExpressionTree tree) {
         Type funcInterfaceType = ((JCTree.JCLambda) tree).type;
+        Queue<Type> allSupertypes = new ArrayDeque<Type>();
+        allSupertypes.add(funcInterfaceType);
         // we want the method symbol for the single function inside the interface...hrm
-        List<Symbol> enclosedElements = funcInterfaceType.tsym.getEnclosedElements();
+        List<Symbol> enclosedElements;
 
         Symbol.MethodSymbol result = null;
-        for (Symbol s: enclosedElements) {
-            Symbol.MethodSymbol elem = (Symbol.MethodSymbol) s;
-            if (elem.isDefault() || elem.isStatic()) {
-                continue;
+        do {
```

Comment:
Can we clean this up by passing in a `com.sun.tools.javac.code.Types` object and then using `Types.closure()`?  E.g. see [this code](https://github.com/google/error-prone/blob/master/check_api/src/main/java/com/google/errorprone/util/ASTHelpers.java#L503).  That way we don't need to explicitly traverse to supertypes, check for duplicates, etc.

Reply:
Sounds good to me. Fixed now :)